bench: refactor random value generation

saurabhraghuvanshii opened the pull request:
Resolves none.

## Description

> What is the purpose of this pull request?

This pull request:

-   refractor random number generation in `math/base/special/binet*`

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves none

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

saurabhraghuvanshii committed:
bench: refactor random generation

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/binet/benchmark/benchmark.js:
```diff
@@ -35,10 +34,11 @@ bench( pkg, function benchmark( b ) {
 	var y;
 	var i;
 
+	x = discreteUniform( 100, 0.0, 79.0 );
```
```suggestion
	x = discreteUniform( 100, 0, 79 );
```

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/binet/benchmark/benchmark.native.js:
```diff
@@ -44,10 +43,11 @@ bench( pkg+'::native', opts, function benchmark( b ) {
 	var y;
 	var i;
 
+	x = discreteUniform( 100, 0.0, 79.0 );
```
Same comment and applies throughout the PR.

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/binet/benchmark/c/benchmark.c:
```diff
@@ -16,6 +16,7 @@
 * limitations under the License.
 */
 
+#include "stdlib/math/base/special/binet.h"
```
Thanks for catching that!

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/binet/benchmark/c/benchmark.c:
```diff
@@ -99,16 +100,19 @@ double binet( double x ) {
 	for ( i = 0; i < ITERATIONS; i++ ) {
-		x = floor( 79.0*rand_double() );
-		y = binet( x );
+		y = stdlib_base_binet( x[ i % 100 ] );
```
```suggestion
		y = stdlib_base_binet( x[ i%100 ] );
```

Minor change just to be consistent with other refactoring related PRs. This change applies throughout the PR.

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/binet/benchmark/c/native/benchmark.c:
```diff
@@ -90,16 +90,19 @@ static double rand_double( void ) {
 	for ( i = 0; i < ITERATIONS; i++ ) {
-		x = floor( 79.0 * rand_double() );
-		y = stdlib_base_binet( x );
+		y = stdlib_base_binet( x[ i % 100 ] );
```
Same comment.

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/binet/test/test.js:
```diff
@@ -41,19 +41,19 @@ tape( 'main export is a function', function test( t ) {
 tape( 'if provided `NaN`, the function returns `NaN`', function test( t ) {
 	var v = binet( NaN );
-	t.strictEqual( isnan( v ), true, 'returns NaN when provided a NaN' );
+	t.strictEqual( isnan( v ), true, 'returns expected value when provided a NaN' );
```
```suggestion
	t.strictEqual( isnan( v ), true, 'returns expected value' );
```

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/boxcox/benchmark/benchmark.js:
```diff
@@ -34,12 +34,15 @@ bench( pkg, function benchmark( b ) {
-		x = ( randu()*10.0 ) + 1.0;
-		y = ( randu()*10.0 ) + 1.0;
-		r = boxcox( x, y );
+		j = i % x.length;
```
`j` is not needed here. Please follow the same refactoring method as has been done already.

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/boxcox/benchmark/benchmark.native.js:
```diff
@@ -43,12 +43,15 @@ bench( pkg+'::native', opts, function benchmark( b ) {
-		x = ( randu() * 10.0 ) + 1.0;
-		y = ( randu() * 10.0 ) + 1.0;
-		r = boxcox( x, y );
+		j = i % x.length;
```
Same comment.

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/boxcox1p/benchmark/benchmark.js:
```diff
@@ -34,12 +34,15 @@ bench( pkg, function benchmark( b ) {
-		x = randu() * 10.0;
-		y = randu() * 10.0;
-		r = boxcox1p( x, y );
+		j = i % x.length;
```
Same comment

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/boxcox1p/benchmark/benchmark.native.js:
```diff
@@ -43,12 +43,15 @@ bench( pkg+'::native', opts, function benchmark( b ) {
-		x = randu() * 10.0;
-		y = randu() * 10.0;
-		r = boxcox1p( x, y );
+		j = i % x.length;
```
Same comment and applies throughout the PR.

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/boxcoxinv/benchmark/benchmark.js:
```diff
@@ -34,12 +34,15 @@ bench( pkg, function benchmark( b ) {
+	var j;
+
+	x = uniform( 100, 0.0, 10.0 );
+	y = uniform( x.length, 0.0, 10.0 );
```
```suggestion
	y = uniform( 100, 0.0, 10.0 );
```

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/boxcoxinv/benchmark/benchmark.native.js:
```diff
@@ -43,12 +43,15 @@ bench( pkg+'::native', opts, function benchmark( b ) {
+	var j;
+
+	x = uniform( 100, 0.0, 10.0 );
+	y = uniform( x.length, 0.0, 10.0 );
```
Same comment and applies throughout the PR.

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/boxcox1pinv/benchmark/c/native/benchmark.c:
```diff
@@ -90,18 +90,21 @@ static double rand_double( void ) {
 	int i;
 
+	for ( i = 0; i < 100; i++ ) {
+		x[ i ] = ( rand_double() * 10.0 );
+		y[ i ] = ( rand_double() * 10.0 );
```
```suggestion
		x[ i ] = rand_double() * 10.0;
		y[ i ] = rand_double() * 10.0;
```

anandkaranubc replied:
This change also applies throughout the PR.

anandkaranubc reviewed (changes requested):
This PR still requires a lot of changes in order to be merged.

saurabhraghuvanshii committed:
Merge branch 'stdlib-js:develop' into random-math1

saurabhraghuvanshii committed:
chore: updates values

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: failed
---

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/boxcoxinv/benchmark/benchmark.js:
```diff
@@ -34,15 +34,13 @@ bench( pkg, function benchmark( b ) {
 
-	x = uniform( 100, 0.0, 10.0 );
-	y = uniform( x.length, 0.0, 10.0 );
+	x = uniform( 100, 0, 10 );
```
This should be `x = uniform( 100, 0.0, 10.0 );`

This would help: 

https://github.com/aayush0325/stdlib/blob/71817278b22a8d65e0ff90f6425c7939a4ee6269/docs/contributing/FAQ.md#when-should-i-use-decimals-in-examples-benchmarks-and-documentation-and-when-should-i-avoid-them

anandkaranubc replied:
This is expected throughout the PR.

saurabhraghuvanshii replied:
one thing I don't understand when should I put `.0` and when not It will be help full for me

saurabhraghuvanshii replied:
sorry I read it .

saurabhraghuvanshii committed:
chore: update values

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

saurabhraghuvanshii committed:
Merge branch 'stdlib-js:develop' into random-math1

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/boxcox/benchmark/benchmark.js:
```diff
@@ -35,11 +35,12 @@ bench( pkg, function benchmark( b ) {
 	var r;
 	var i;
 
+	x = uniform( 100, 1, 11 );
+	y = uniform( 100, 1, 11 );
```
```suggestion
	x = uniform( 100, 1.0, 11.0 );
	y = uniform( 100, 1.0, 11.0 );
```

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/boxcox/benchmark/benchmark.native.js:
```diff
@@ -44,11 +44,12 @@ bench( pkg+'::native', opts, function benchmark( b ) {
 	var r;
 	var i;
 
+	x = uniform( 100, 1, 11 );
+	y = uniform( 100, 1, 11 );
```
Same comment

saurabhraghuvanshii committed:
chore: improve precision

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

saurabhraghuvanshii commented:
/stdlib help

saurabhraghuvanshii commented:
/stdlib merge

stdlib-bot committed:
Merge remote-tracking branch 'upstream/develop' into random-math1

anandkaranubc reviewed (approved):
The PR looks ready to be merged. Thanks for making those changes!

saurabhraghuvanshii commented:
I have to thanks you for reviewing my commits . I try to improve more .

Planeshifter reviewed (approved):
Let's get this one in. 🎉 

Thanks @saurabhraghuvanshii for your work on this and @anandkaranubc for leaving review comments; much appreciated!
